src/wallet/wallet.cpp

```diff
@@ -2534,8 +2534,17 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
                     return false;
                 }
 
-                if (nFeeRet >= nFeeNeeded)
+                if (nFeeRet >= nFeeNeeded) {
+                    // Reduce fee to only the needed amount if we have change output to increase
+                    // Do not do this if fee was paid by payee
+                    if (nFeeRet > nFeeNeeded && nChangePosInOut != -1 && nSubtractFeeFromAmount == 0) {
```

Comment:
it's pretty easy to compute exactly the amount of additional fee adding a change output would imply. So the test becomes if not subtractfrom   and excessfee >= CENT/2 + changeaddfee; then set the change to excessfee-changeaddfee.   Doing so without creating a hairball with change added in multiple places... less fun.  

Similarly, the remove side could consider eliminating the change output, if doing so would not overpay by too much. (A conservative value would be the changeaddfee but a larger value would be a lot more useful)  both both probably better done in another PR.